### DISCUSSION TITLE
Automatically time beatmaps on song import

### DISCUSSION THREAD

**Discussion by Xinnoh** [ppy/osu#17717] [Ideas]
When importing a song, Malody automatically times beatmaps and gives bpm, offset, and uncertainty level.
This feature would save so many new mappers from headaches, and also save experienced mappers a lot of time.

This seems like such an obvious and important feature to have but I've never seen it brought up much

![Screenshot 2022-04-07 173800](https://user-images.githubusercontent.com/30382015/162341184-d1856a66-ba0f-43be-9e49-028766a0386a.jpg)

**Comment by peppy** [comment]
It already existed in stable but was not reliable enough to consider it a good idea (and so I removed it). We may revisit in the future, but it's *very* hard to get the offset correct automatically.

I can agree BPM detection may have a place though. Algorithms have come a long way.

**Reply by Ceynou** [reply]
You could implement BPM and offset detection based on this [paper](https://github.com/nathanstep55/bpm-offset-detector/blob/main/original-paper/report.pdf) (made by the developper of arrowvortex editor)
Set it by default because I'm almost sure it's more accurate than most if not all the mappers.
Maybe lock setting it manually with a message "Only edit this if you know what you're doing"
